lib/image-stream/archive.js

```diff
@@ -81,8 +80,20 @@ const extractEntryByPath = (archive, filePath, options) => {
     return Bluebird.resolve(options.default);
   }
 
-  return options.hooks.extractFile(archive, options.entries, fileEntry.name)
-    .then(rindle.extract);
+  return options.hooks.extractFile(archive, options.entries, fileEntry.name).then((stream) => {
+    return new Bluebird((resolve, reject) => {
+      let data = '';
```

Comment:
Afaik, this is for extracting metadata files from archives, not the images – which means that memory exhaustion is not a concern (and before it would be, you'll get a `RangeError: Array buffer allocation failed`). The images themselves are streamed.